File: api_service/src/services/auth/downstream_grant_svc.py
from src.models.auth.auth import DownstreamAccessGrant
from src.models.auth.auth_contract import DownstreamGrantRequest
from src.services.auth.auth_authority import IAuthAuthorityClient


class DownstreamGrantService:
    def __init__(self, authority_client: IAuthAuthorityClient):
        self._authority_client = authority_client

    async def issue_downstream_grant(
        self,
        req: DownstreamGrantRequest,
    ) -> DownstreamAccessGrant:
        return await self._authority_client.issue_downstream_grant(req)

File: certification_server/src/iface/auth/downstream_grant_svc.go
package auth

import (
	authmodel "certification_server/src/models/auth"
	"context"
)

// IDownstreamGrantService 定义下游服务访问授权签发接口。
type IDownstreamGrantService interface {
	// IssueDownstreamGrant 根据认证上下文和目标服务签发下游访问授权。
	IssueDownstreamGrant(ctx context.Context, req *authmodel.DownstreamGrantRequest) (*authmodel.DownstreamAccessGrant, error)
}
